### PR TITLE
Remove setup from the containerized server: moved to uyuniadm

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Remove the setup for containerized server and the systemd service RPM as this is now done by uyuniadm.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: Not for `master` yet

- [X] **DONE**

## Test coverage
- No tests: tested with the E2E tests

- [X] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by cbosdo** [comment]
@uyuni-project/release-engineering you can remove yourself from this PR: not for master.

Note that the README still needs some love.

**Review by mbussolotto — APPROVED** [review]

File: containers/doc/server-kubernetes/README.md [hunk 1]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
 ```
 
-Once done, both the job and its pod will remain until the user deletes them to allow checking logs.
-
-Proceed with the next steps.
+# Migrating from a regular server
 
-***Hostname***: this procedure doesn't handle any hostname change.
-Certificates migration also needs to be documented, but that can be guessed for now with the instructions to setup a server from scratch.
+The migration procedure does not perform any hostname rename.
```
Comment (rjmateus) [inline containers/doc/server-kubernetes/README.md:107]: I think you should mention in here that we cannot perform an in-place migration, and a new machine is needed.

**Review by rjmateus — APPROVED** [review]
Just one small note

File: containers/doc/server-kubernetes/README.md [hunk 2]
```diff
@@ -1,13 +1,15 @@
-# Running the server-image on kubernetes
+# Prerequisites
 
-## Prerequisites
+The following assumes you have either a single-node rke2 or k3s cluster ready or a server with `podman` installed and enough resources for the Uyuni server.
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:3]:
```suggestion
The following assumes you have either a single-node RKE2 or K3s cluster ready or a server with `podman` installed and enough resources for the Uyuni server.
```

File: containers/doc/server-kubernetes/README.md [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
 
-## Setting up the resources
+## Podman specific setup
 
-### RKE2 specific setup
+There is nothing to prepare for a `podman` installation.
+
+## RKE2 specific setup
 
 Copy the `rke2-ingress-nginx-config.yaml` file to `/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml` on your rke2 node.
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:14]:
```suggestion
Copy the `rke2-ingress-nginx-config.yaml` file to `/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml` on your RKE2 node.
```

File: containers/doc/server-kubernetes/README.md [hunk 4]
```diff
@@ -17,7 +19,7 @@ Run this command to watch it restart:
 watch kubectl get -n kube-system pod -lapp.kubernetes.io/name=rke2-ingress-nginx
 ```
 
-### K3s specific setup
+## K3s specific setup
 
 
 Copy the `k3s-traefik-config.yaml` file to `/var/lib/rancher/k3s/server/manifests/` on your k3s node.
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:25]:
```suggestion
Copy the `k3s-traefik-config.yaml` file to `/var/lib/rancher/k3s/server/manifests/` on your K3s node.
```

File: containers/doc/server-kubernetes/README.md [hunk 5]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
 watch kubectl get -n kube-system pod -lapp.kubernetes.io/name=traefik
 ```
 
-***Offline installation:*** with k3s it is possible to preload the container images and avoid it to be fetched from a registry.
+# Offline installation
+
+
+## For k3s
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:36]:
```suggestion
## For K3s
```

File: containers/doc/server-kubernetes/README.md [hunk 6]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
+
+With k3s it is possible to preload the container images and avoid it to be fetched from a registry.
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:38]:
```suggestion
With K3s it is possible to preload the container images and avoid it to be fetched from a registry.
```

File: containers/doc/server-kubernetes/README.md [hunk 7]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
 or
 
+⚠️ **TODO**: instructions to pull `cert-manager` images
 ```
 skopeo copy docker://registry.opensuse.org/systemsmanagement/uyuni/master/servercontainer/containers/uyuni/server:latest docker-archive:server-image.tar:registry.opensuse.org/systemsmanagement/uyuni/master/servercontainer/containers/uyuni/server:latest
 ```
 
+⚠️ **TODO**: also copy the `cert-manager` and `uyuni/server` helm charts locally
+
 Transfer the resulting `server-image.tar` to the k3s node and load it using the following command:
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:57]:
```suggestion
Transfer the resulting `server-image.tar` to the K3s node and load it using the following command:
```

File: containers/doc/server-kubernetes/README.md [hunk 8]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
 
 ```
 k3s ctr images import server-image.tar
 ```
 
-In order to tell k3s to not pull the image, add `imagePullPolicy: Never` to all `initContainer`s and `container` in the `server.yaml` file:
+In order to tell k3s to not pull the image, set the `pullPolicy` helm chart value to `Never` by passing a `--helm-value=values.yaml` parameter to `uyuniadm` with the following `values.yaml` file content:
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:63]:
```suggestion
In order to tell K3s to not pull the image, set the `pullPolicy` helm chart value to `Never` by passing a `--helm-value=values.yaml` parameter to `uyuniadm` with the following `values.yaml` file content:
```

File: containers/doc/server-kubernetes/README.md [hunk 9]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
 
 ```
-sed 's/^\( \+\)image:\(.*\)$/\1image: \2\n\1imagePullPolicy: Never/' -i server.yaml
+pullPolicy: Never
 ```
 
-### Migrating from a regular server
+## For rke2
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:69]:
```suggestion
## For RKE2
```

File: containers/doc/server-kubernetes/README.md [hunk 10]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
+# TODO copy the cert-manager and uyuni helm charts
+# TODO Copy the cert-manager and uyuni images
+# TODO Install cert-manager from the local chart
+# TODO Set the uyuniadm --image parameter
 ```
 
-Create a password-less SSH key and create a kubernetes secret with it:
+## For podman
+
+With k3s it is possible to preload the container images and avoid it to be fetched from a registry.
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:84]:
```suggestion
With K3s it is possible to preload the container images and avoid it to be fetched from a registry.
```

File: containers/doc/server-kubernetes/README.md [hunk 11]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
+Run the following command to install a new Uyuni server from the source one after replacing the `uyuni.source.fqdn` by the proper source server FQDN:
+This command will synchronize all the data from the source server to the new one: this can take time!
 
 ```
-kubectl create configmap uyuni-ca --from-file=ca.crt=/var/uyuni/ssl-build/RHN-ORG-TRUSTED-SSL-CERT
+uyuniadm migrate uyuni.source.fqdn
 ```
 
-### CA certificates using Cert-Manager
+## Notes for kubernetes
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:147]:
```suggestion
## Notes for Kubernetes
```

File: containers/doc/server-kubernetes/README.md [hunk 12]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
 
-For security reason, copy the CA certificate into a separate config map, to not mount the CA secret on the pod:
+## Volumes preparation 
 
-```
-kubectl get secret uyuni-ca -o=jsonpath='{.data.ca\.crt}' | base64 -d >ca.crt
-kubectl create configmap uyuni-ca --from-file=ca.crt
-rm ca.crt
-```
+### For kubernetes
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:160]:
```suggestion
### For Kubernetes
```

File: containers/doc/server-kubernetes/README.md [hunk 13]
```diff
@@ -1,13 +1,15 @@
-# Running the server-image on kubernetes
+# Prerequisites
 
-## Prerequisites
+The following assumes you have either a single-node rke2 or k3s cluster ready or a server with `podman` installed and enough resources for the Uyuni server.
+When installing on a kubernetes cluster, it also assumes that `kubectl` and `helm` are installed on the server and configured to connect to the cluster.
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:4]:
```suggestion
When installing on a Kubernetes cluster, it also assumes that `kubectl` and `helm` are installed on the server and configured to connect to the cluster.
```

File: containers/doc/server-kubernetes/README.md (hunk 3 above)
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:14]:
```suggestion
Copy the `rke2-ingress-nginx-config.yaml` file to `/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml` on your RKE2 node.
```

File: containers/doc/server-kubernetes/README.md (hunk 4 above)
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:25]:
```suggestion
Copy the `k3s-traefik-config.yaml` file to `/var/lib/rancher/k3s/server/manifests/` on your K3s node.
```

File: containers/doc/server-kubernetes/README.md [hunk 14]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
-spacewalk-service stop
-systemctl stop postgresql
+# TODO copy the cert-manager and uyuni helm charts
+# TODO Copy the cert-manager and uyuni images
+# TODO Install cert-manager from the local chart
+# TODO Set the uyuniadm --image parameter
 ```
 
-Create a password-less SSH key and create a kubernetes secret with it:
+## For podman
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:82]:
```suggestion
## For Podman
```

File: containers/doc/server-kubernetes/README.md [hunk 15]
```diff
@@ -28,143 +30,144 @@ Run this commant to watch it restart:
...
-Run the following command to append the ingress annotation to use the new CA when applying the helm chart later:
+⚠️ **TODO** Document this
 
-```
-cat >values.yaml << EOF
-ingressSslAnnotations:
-  cert-manager.io/issuer: uyuni-ca-issuer
-EOF
-```
+### For podman
```
Comment (deneb-alpha) [inline containers/doc/server-kubernetes/README.md:164]:
```suggestion
### For Podman
```

**Review by deneb-alpha — CHANGES_REQUESTED** [review]
Added some suggestions for fixing the case of:
* RKE2
* K3s
* Podman
* Kubernetes

**Pull request merged**